src/hooks/useSourceApi.ts

```diff
@@ -1,105 +1,329 @@
+      });
+
+      return;
+    },
+    [onAddAlert, t]
+  );
+
+  const callbackError = useCallback(
+    ({ message, response }: AxiosError<ApiDeleteSourceErrorType>, updatedSources: SourceType[]) => {
+      const errorMessage = t('toast-notifications.description', {
+        context: 'deleted-source_error',
+        count: updatedSources.length,
+        name: updatedSources.map(({ name }) => name).join(', '),
+        message: response?.data?.detail || response?.data?.message || message || 'Unknown error'
+      });
+      onAddAlert({ title: errorMessage, variant: 'danger', id: helpers.generateId() });
+    },
+    [onAddAlert, t]
+  );
+
+  const deleteSources = useCallback(
+    async (source: SourceType | SourceType[]) => {
+      const updatedSources = (Array.isArray(source) && source) || [source];
+      try {
+        const response = await apiCall(updatedSources.map(({ id }) => id));
+        callbackSuccess(response, updatedSources);
+      } catch (error) {
+        if (isAxiosError(error)) {
+          return callbackError(error, updatedSources);
+        }
+        if (!helpers.TEST_MODE) {
+          console.error(error);
+        }
+      }
+    },
+    [apiCall, callbackSuccess, callbackError]
+  );
+
+  return {
+    apiCall,
+    callbackError,
+    callbackSuccess,
+    deleteSources
   };
+};
 
-  /**
-   * Executes a GET request to retrieve connection information related to a source.
-   *
-   * @param {SourceType} source - The source to which connections are related.
-   * @returns {AxiosResponse} - The Axios response object representing the result of the request.
-   */
-  const showConnections = (source: SourceType) => {
-    return axios.get(
-      `${process.env.REACT_APP_SCAN_JOBS_SERVICE}${source.connection.id}/connection/?page=1&page_size=1000&ordering=name&source_type=${source.id}`
-    );
+const useAddSourceApi = (onAddAlert: (alert: AlertProps) => void) => {
+  const { t } = useTranslation();
+
+  const apiCall = useCallback(
+    (payload: SourceType): Promise<AxiosResponse<SourceType>> =>
+      axios.post(`${process.env.REACT_APP_SOURCES_SERVICE}?scan=true`, payload),
+    []
+  );
+
+  const callbackSuccess = useCallback(
+    (response: AxiosResponse<SourceType>) => {
+      const successMessage = t('toast-notifications.description', {
+        context: 'add-source_hidden',
+        name: response.data.name
+      });
+      onAddAlert({
+        title: successMessage,
+        variant: 'success',
+        id: helpers.generateId()
+      });
+
+      return;
+    },
+    [onAddAlert, t]
+  );
+
+  const callbackError = useCallback(
+    (error: AxiosError<ApiDeleteSourceErrorType>, name: string) => {
+      const errorMessage = t('toast-notifications.title', {
+        context: 'add-source_hidden_error',
+        name: name,
+        message: helpers.extractErrorMessage(error?.response?.data)
+      });
+      onAddAlert({ title: errorMessage, variant: 'danger', id: helpers.generateId() });
```

Comment:
required:
minor styling adjustment, similar to before. move `errorMessage` directly under `onAddAlert`

```
onAddAlert({
  title: t('toast-notifications.title', {
    context: 'add-source_hidden_error',
    name: name,
    message: helpers.extractErrorMessage(error?.response?.data)
  }),
  variant: 'danger',
  id: helpers.generateId()
});
```